app/test-utils/testFunctions/questionTests.js

```diff
@@ -0,0 +1,93 @@
+import { runRadioButtonTests } from './questionTests/radioButtonTests';
+import { runtextAreaTests } from './questionTests/textAreaTests';
+import { runCheckboxTests } from './questionTests/checkboxTests';
+import { runTextFieldTests } from './questionTests/textFieldTests';
+import { runBulletpointListTests } from './questionTests/bulletpointListTests';
+import { runErrorTests } from './questionTests/errorTests';
+import { runComboboxTests } from './questionTests/comboboxTests';
+
+export const runQuestionTests = ({
+  pageSetup,
+  sectionManifest,
+  sectionApiUrl,
+  data,
+  sectionId,
+  apiLocalhost,
+  sectionParent = null,
+}) => {
+  Object.keys(sectionManifest.questions).forEach(async (questionId) => {
+    const questionData = sectionManifest.questions[questionId];
+    const questionType = questionData.type;
+    if (questionType === 'radiobutton-options') {
+      runRadioButtonTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+      });
+    } else if (questionType === 'textarea-field') {
+      runtextAreaTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+      });
+    } else if (questionType === 'checkbox-options') {
+      runCheckboxTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+      });
+    } else if (questionType === 'text-field') {
+      runTextFieldTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+        sectionParent,
+      });
+    } else if (questionType === 'bulletpoint-list') {
+      runBulletpointListTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+      });
+    } else if (questionType === 'multi-question') {
+      Object.keys(sectionManifest.questions).forEach((question) => {
+        runQuestionTests({
+          pageSetup,
+          sectionManifest: sectionManifest.questions[question],
+          sectionApiUrl,
+          data,
+          sectionId,
+          apiLocalhost,
+          sectionParent: question,
+        });
+      });
+    } else if (questionType === 'combobox-options') {
+      runComboboxTests({
+        pageSetup,
+        sectionManifest,
+        questionId,
+        data,
+      });
+    } else {
```

Comment:
Not one for now but could we have a map based upon the questionType and then a function it does to map to the function you want to run. There is something similar in `createQuestionsContext` that might be helpful

Reply:
there's probably loads of refactoring to be done - it grew to be like this!

Reply:
Cool we can sort this out later